packages/web/src/pages/dashboard-page/components/ArtistContentSection.tsx

```diff
@@ -0,0 +1,170 @@
+    (!isTracks && !hasOnlyOneAlbumSection)
+  const shouldShowPills = tracks.length && albums.length
+
+  const onClickRow = useCallback(
+    (record: any) => {
+      if (!account) return
+      goToRoute(record.permalink)
+    },
+    [account, goToRoute]
+  )
+
+  const onClickPill = useCallback(
+    (pill: Pills) => {
+      setSelectedPill(pill)
+      setFilterText('')
+
+      // Reset filter button state when switching content types
+      if (!isTracks && pill === Pills.TRACKS) {
+        setSelectedAlbumFilter(null)
+      } else if (isTracks && pill === Pills.ALBUMS) {
+        setSelectedTrackFilter(null)
+      }
+    },
+    [isTracks]
+  )
+
+  const handleSelectFilter = (value: string) => {
+    if (isTracks) {
+      setSelectedTrackFilter(value as TrackFilters)
+    } else {
+      setSelectedAlbumFilter(value as AlbumFilters)
+    }
+  }
+
+  const handleFilterChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const val = e.target.value
+    setFilterText(val)
+  }
+
+  if (!tracks.length && !albums.length) return null
+
+  return (
+    <Paper w='100%' direction='column' mt='xl'>
+      <Flex ph='2xl' pv='l' justifyContent='space-between'>
+        <Flex gap='2xl'>
+          {shouldShowPills ? (
+            <Flex gap='s'>
+              <SelectablePill
+                isSelected={selectedPill === Pills.TRACKS}
+                label={messages.tracks}
+                size='large'
+                onClick={() => onClickPill(Pills.TRACKS)}
+              />
+              <SelectablePill
+                isSelected={selectedPill === Pills.ALBUMS}
+                label={messages.albums}
+                size='large'
+                onClick={() => onClickPill(Pills.ALBUMS)}
+              />
+            </Flex>
+          ) : null}
+          {
+            // Only show filter button if there are multiple sections for the selected content type
+            shouldShowFilterButton ? (
+              <FilterButton
+                onSelect={handleSelectFilter}
+                selection={isTracks ? selectedTrackFilter : selectedAlbumFilter}
+                label={messages.allReleases}
+                options={filterButtonOptions}
+                popupAnchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
+                popupTransformOrigin={{ vertical: 'top', horizontal: 'left' }}
+              />
+            ) : null
+          }
+        </Flex>
+        <Flex>
+          <TextInput
+            placeholder={messages.search(selectedPill)}
+            label={messages.search(selectedPill)}
+            value={filterText}
+            onChange={handleFilterChange}
+            size={TextInputSize.SMALL}
+            startIcon={IconSearch}
+          />
+        </Flex>
+      </Flex>
+      {selectedPill === Pills.TRACKS ? (
+        <ArtistDashboardTracksTab
+          selectedFilter={selectedTrackFilter}
+          filterText={filterText}
+          onClickRow={onClickRow}
```

Comment:
nit: This could also be pushed into the table if there is no logic to be done or data to be coordinated at this level. Since it's just navigating on click, I think you could build that into the child components so they are self-cointained minus the filter text.